amazon-redshift-plugin/src/main/java/io/cdap/plugin/amazon/redshift/RedshiftSchemaReader.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.amazon.redshift;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.plugin.db.CommonSchemaReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.List;
+import java.util.Set;
+
+/**
+ *    Redshift Schema Reader class
+ */
+public class RedshiftSchemaReader extends CommonSchemaReader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RedshiftSchemaReader.class);
+
+  public static final Set<Integer> STRING_MAPPED_REDSHIFT_TYPES = ImmutableSet.of(
```

Comment:
Check the documentation here for Java type conversion: https://docs.aws.amazon.com/redshift/latest/mgmt/jdbc20-data-type-mapping.html

Why are we mapping OTHER, ARRAY and SQLXML types? I don't think these are even present in Redshift. Was this tested?

Reply:
Redshift is postgres based DB, some of the array features are moved to Super. We have tested with all data types that redshift allows. Removing this also.